### PR TITLE
spiral-matrix: implement.

### LINKED ISSUE #752: spiral-matrix: implement exercise
There are exercises that are not yet implemented in Python track (the full list of them - https://github.com/exercism/python/issues/417#issuecomment-334841778).
You can help us change this situation by implementing this one. Ref #417

The guide ["Implement an exercise from specification"](https://github.com/exercism/docs/blob/master/you-can-help/implement-an-exercise-from-specification.md) will help you succeed.

If you still have questions, I encourage you to ask them before starting working on the issue.

I also suggest claiming the issue by creating an empty PR as stated [here](https://github.com/exercism/docs/blob/master/you-can-help/implement-an-exercise-from-specification.md#avoiding-duplicate-work)

Please note that there are some special ['closing issue' keywords](https://help.github.com/articles/closing-issues-using-keywords/) that you can put in description of your PR and Github will take care of closing the corresponding issue automatically after merge.
This simple thing helps us a lot, and makes our life a bit easier

### PR DESCRIPTION
**Pull request by Average-user**
It is on the Unimplemented exercises list of Python in [exercism.io](exercism.io). So, is better if it can stop being there.

----
Resolves #752

### CODE REVIEW

File: exercises/spiral-matrix/spiral_matrix.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+def spiral():
```
Comment (m-a-ge) [inline exercises/spiral-matrix/spiral_matrix.py]: It was reported here https://github.com/exercism/python/issues/509 that "placeholder" files should have expected parameters to help users start working on exercises quicker

File: config.json [hunk 2]
```diff
@@ -725,6 +725,15 @@
       ]
     },
     {
+      "uuid": "666fda78-851e-4bb0-a416-953528f46bd7",
+      "slug": "spiral-matrix",
+      "core": false,
+      "unlocked_by": null,
+      "difficulty": 2,
+      "topics": [
```
[the remark this answers is not in the document]
Comment (Average-user) [reply config.json]: i'm not sure what to put. some idea ?

**Comment by Average-user** [comment]
I'm not sure about the topic, feel free to propose anything about them

File: config.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (m-a-ge) [reply config.json]: @Average-user I guess the following ones will be Ok:
```
loops
lists
```
`algorithms` is also suitable

File: exercises/spiral-matrix/spiral_matrix_test.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import unittest
+
+from spiral_matrix import spiral
+
```
Comment (m-a-ge) [inline exercises/spiral-matrix/spiral_matrix_test.py]: Could you add a comment that this tests are based on version `1.0.0` of https://github.com/exercism/problem-specifications/blob/master/exercises/spiral-matrix/canonical-data.json?

File: exercises/spiral-matrix/spiral_matrix.py [hunk 4]
```diff
@@ -0,0 +1,2 @@
+def spiral(n):
```
Comment (m-a-ge) [inline exercises/spiral-matrix/spiral_matrix.py]: Could you please change `n` to something more meaningful like `matrix_size`?

**Comment by N-Parsons** [comment]
@Average-user, are you still working on this?

**Comment by Average-user** [comment]
I thought it was ready, I'll see whats left

**Comment by Average-user** [comment]
@m-a-ge  I think there is nothing  left to do. Right?

File: exercises/spiral-matrix/spiral_matrix_test.py [hunk 5]
```diff
@@ -5,6 +5,7 @@
 # Tests are based on the version 1.0.0 of:
 # https://github.com/exercism/problem-specifications/blob/master/exercises/spiral-matrix/canonical-data.json
```
Comment (cmccandless) [inline exercises/spiral-matrix/spiral_matrix_test.py]: Per #784, the format of the canonical version comment should be as follows:
```Python
# Tests adapted from `problem-specifications//canonical-data.json` @ v1.0.0
```

File: config.json [hunk 6]
```diff
@@ -725,6 +725,18 @@
...
+        "algorithms",
+        "control-flow (loops)",
```
Comment (N-Parsons) [inline config.json]: `control-flow (loops)` should just be `loops` - see [TOPICS.txt](https://github.com/exercism/problem-specifications/blob/master/TOPICS.txt) for a full list of valid topics.

File: exercises/spiral-matrix/README.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+###### Spiral matrix of size 4
+
+```plain
+ 1  2  3 4
+12 13 14 5
+11 16 15 6
+10  9  8 7
+```
+## Source
```
Comment (N-Parsons) [inline exercises/spiral-matrix/README.md]: There should be a blank line between sections, and there should actually be a `## Submitting exercises` section here.

File: exercises/spiral-matrix/spiral_matrix_test.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
+import unittest
+
+from spiral_matrix import spiral
+
+# Tests are based on the version 1.0.0 of:
+# https://github.com/exercism/problem-specifications/blob/master/exercises/spiral-matrix/canonical-data.json
```
Comment (N-Parsons) [inline exercises/spiral-matrix/spiral_matrix_test.py]: This comment should be ```# Tests adapted from `problem-specifications//canonical-data.json` @ v1.0.0``` as per #784. Could you also move it down by a line? (ie. two blank lines before, one after)

Comment (Average-user) [reply exercises/spiral-matrix/spiral_matrix_test.py]: You want me to put something between those two slashes, or not?

Comment (N-Parsons) [reply exercises/spiral-matrix/spiral_matrix_test.py]: @Average-user, no, it's just a shorthand way of saying that we are skipping some uninteresting directories in the middle. The full path would be `problem-specifications/exercises/spiral-matrix/canonical-data.json`, which is a bit long to include in a comment, and the middle directories are fairly obvious.

**Comment by N-Parsons** [comment]
@Average-user, are you still working on this? There are just a few minor changes that need to be made before this can be merged.

**Pull request closed without merging**